pkg/arvo/app/landscape/js/channel.js

```diff
@@ -159,21 +187,29 @@ class Channel {
     req.setRequestHeader("Content-Type", "application/json");
 
     if (this.lastEventId == this.lastAcknowledgedEventId) {
-      let x = JSON.stringify([j]);
+      if(j) {
+        this.outstandingJSON.push(j);
+      }
+      let x = JSON.stringify(this.outstandingJSON);
       req.send(x);
+      this.outstandingJSON = [];
     } else {
       //  we add an acknowledgment to clear the server side queue
       //
       //    The server side puts messages it sends us in a queue until we
       //    acknowledge that we received it.
       //
-      let payload = [{action: "ack", "event-id": parseInt(this.lastEventId)}];
+      let payload = [
+        ...this.outstandingJSON, 
+        {action: "ack", "event-id": parseInt(this.lastEventId)}
+      ];
       if(j) {
         payload.push(j)
       }
       let x = JSON.stringify(payload);
       req.send(x);
 
+      this.outstandingJSON = [];
       this.lastEventId = this.lastAcknowledgedEventId;
```

Comment:
Might be better to factor this to come after/outside of the if-else here, since it's now (even more obviously) shared logic between the two branches.